### PR TITLE
Chore: create a release when a PR is merged to main

### PR DESCRIPTION
**Pull request by katspaugh**
## What it solves

I've added an action to automatically tag a merged RC and make a release with the same text as the release PR.

I've tested the new action in a private repo.

### CODE REVIEW

File: .github/workflows/tag-release.yml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+        if: github.event.pull_request.merged == true
+        with:
+          fetch-depth: 0
+
+      - name: Extract version
+        if: github.event.pull_request.merged == true
+        id: version
+        run: |
+          NEW_VERSION=$(node -p 'require("./package.json").version')
+          echo "::set-output name=version::v$NEW_VERSION"
```
Comment (usame-algan) [inline .github/workflows/tag-release.yml:24]: Would this be the same as `echo "version=v$NEW_VERSION" >> $GITHUB_OUTPUT` like described [here](https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#setting-an-output-parameter)?

File: .github/workflows/tag-release.yml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+          echo "${{ github.event.pull_request.body }}" > CHANGELOG.md
+
+      - name: Create a git tag
+        if: github.event.pull_request.merged == true
+        run: git tag $NEW_VERSION && git push --tags
+        env:
+          NEW_VERSION: ${{ steps.version.outputs.version }}
```
Comment (usame-algan) [inline .github/workflows/tag-release.yml:31]: Does `$NEW_VERSION` contain the prefix `v` here even if we set the environment variable after?

Comment (katspaugh) [reply .github/workflows/tag-release.yml:31]: Ah, right, good catch. Let me add a `v`.

File: .github/workflows/tag-release.yml (hunk 1 above)
Comment (katspaugh) [reply .github/workflows/tag-release.yml:24]: I'll try this syntax

File: .github/workflows/tag-release.yml (hunk 2 above)
Comment (katspaugh) [reply .github/workflows/tag-release.yml:31]: Checked, it works correctly like it is.

Comment (usame-algan) [reply .github/workflows/tag-release.yml:31]: Testing this in a private repo it seems to include the prefix

**Review by usame-algan — APPROVED** [review]

**Pull request merged**